### ISSUE TITLE
Investigate if we can build the snaps statically

### ISSUE THREAD

**Issue by bep** [gohugoio/hugo#4534]
This was made by someone in Cannocial:

https://github.com/gohugoio/hugo/blob/master/snapcraft.yaml

Building Hugo as a statically linked binary should fix the issue described in https://discourse.gohugo.io/t/hugos-snap-stopped-working/11184/26

It will create a bigger binary, but it is better than a broken binary.

This is how we build the binaries for the GitHub release:

https://github.com/gohugoio/hugo/blob/master/goreleaser.yml#L6

/cc @elopio @anthonyfok

**Comment by elopio** [comment]
Hey @bep. I'm no longer working on snapcraft, but my friends @popey and @flexiondotorg can give you a hand.

**Comment by anthonyfok** [comment]
Thanks for bringing this to my attention.

And my apologies: I _did_ change something:

I was trying to fix some failed-to-build problems in 0.38-DEV as may be seen here:

* https://launchpad.net/~gohugoio/+snap/hugo-dev
* https://launchpadlibrarian.net/361846747/buildlog_snap_ubuntu_bionic_amd64_hugo-dev_BUILDING.txt.gz

Suspecting that it might had to do with the somewhat outdated go1.6 in Ubuntu Xenial (16.04 LTS), I experimented and eventually discovered that changing the "Distribution series" (from Ubuntu Xenial (16.04 LTS) to Ubuntu Artful (17.10) did the trick, and applied the same to the stable series at https://launchpad.net/~gohugoio/+snap/hugo

Unfortunately, as it turned out, that changed caused the following error as reported in the Hugo discussion forum:

> /snap/hugo/1455/bin/hugo: relocation error: /snap/hugo/1455/lib/x86_64-linux-gnu/libpthread.so.0: symbol __mmap, version GLIBC_PRIVATE not defined in file libc.so.6 with link time reference

which turns out even I can reproduce.  My sincere apologies for causing all the trouble.

I went to https://launchpad.net/~gohugoio/+snap/hugo and changed it back to Ubuntu Xenial, and the latest revision of Hugo snap (e.g. revision 1462 for amd64) works correctly again.  I guess that solves this issue?  :-D
